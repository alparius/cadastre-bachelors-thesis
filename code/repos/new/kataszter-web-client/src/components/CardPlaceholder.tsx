import React from "react";
import { Card, Placeholder } from "semantic-ui-react";

const CardPlaceholder: React.FC = () => {
    return (
        <Card fluid>
            <Card.Content>
                <Card.Header>
                    <Placeholder>
                        <Placeholder.Header>
                            <Placeholder.Line />
                        </Placeholder.Header>
                    </Placeholder>
                </Card.Header>
                <Card.Description>
                    <Placeholder>
                        <Placeholder.Line />
                        <Placeholder.Line />
                        <Placeholder.Line />
                        <Placeholder.Line />
                        <Placeholder.Line />
                        <Placeholder.Line />
                    </Placeholder>
                </Card.Description>
            </Card.Content>
        </Card>
    );
};

export default CardPlaceholder;
