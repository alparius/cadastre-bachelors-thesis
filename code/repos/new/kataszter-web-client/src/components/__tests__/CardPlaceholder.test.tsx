/*===================
  CardPlaceholder.unit.test.tsx
 ===================*/
import React from "react";
import renderer from "react-test-renderer";

import CardPlaceholder from "../CardPlaceholder";

describe("CardPlaceholder element", () => {
    it("should render a card with placeholder lines in it", () => {
        const container = renderer.create(<CardPlaceholder />).toJSON();
        expect(container).toMatchSnapshot();
    });
});
