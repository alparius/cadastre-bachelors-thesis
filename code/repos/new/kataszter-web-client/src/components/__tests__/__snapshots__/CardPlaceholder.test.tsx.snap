// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPlaceholder element should render a card with placeholder lines in it 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="header"
        >
          <div
            className="line"
          />
        </div>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;
